app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -47,6 +47,7 @@ class BrowserToolbarIntegration(
     private val onUrlLongClicked: () -> Boolean,
     private val eraseActionListener: () -> Unit,
     private val tabCounterListener: () -> Unit,
+    private val cfrForShieldToolbarIconListener: () -> Unit,
```

Comment:
Nice separations of concerns and control flow here with using a callback instead of calling methods directly on the fragment which contains this integration.
But the naming is still leaking the expected behavior which should not be known to this class.
Maybe just `onTrackingProtectionShown` and then let the layer above - the fragment decide if this is a good indicator to do some random action like maybe showing a popup?